hkmc2/shared/src/main/scala/hkmc2/semantics/Elaborator.scala

```diff
@@ -820,23 +857,26 @@ extends Importer:
   
   def params(t: Tree): Ctxl[(ParamList, Ctx)] = t match
     case Tup(ps) =>
-      val plf = ParamListFlags.empty
-      def go(ps: Ls[Tree], acc: Ls[Param], ctx: Ctx): (ParamList, Ctx) =
+      def go(ps: Ls[Tree], acc: Ls[Param], ctx: Ctx, flags: ParamListFlags): (ParamList, Ctx) =
         ps match
-        case Nil => (ParamList(plf, acc.reverse, N), ctx)
+        case Nil => (ParamList(flags, acc.reverse, N), ctx)
         case hd :: tl =>
           param(hd)(using ctx) match
           case S((isSpd, p)) =>
+            val isCtx = hd match
+              case Modified(Keyword.`using`, _, _) => true
+              case _ => false
             val newCtx = ctx + (p.sym.name -> p.sym)
+            val newFlags = if isCtx then flags.copy(ctx = true) else flags
```

Comment:
Does this mean that in `(x: Int, using y: Str)`, both `x` and `y` are contextual? What's the point of allowing this confusing syntax?

Reply:
Yes this is what the code currently does. I'm going to fix it (and some other strange behaviors 🫨) later when I convert the draft PR into a PR.